### ISSUE TITLE
Retry when the API is unavailable

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/firefox-ios#15647]
Retry Shopping API after 500 errors

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-7036)

**Comment by data-sync-user** [comment]
➤ Alina Moldovan commented:

Razvan Litianu could you please add some details about the changes the QA team must check for this ticket?

**Comment by data-sync-user** [comment]
➤ Razvan Litianu commented:

So if the api fails we have a requirement to try 3 times more then fail. I’m not sure how QA could see this. It might involve taking the api down and up which is not ideal. This logic in covered by unit testing.

**Comment by data-sync-user** [comment]
➤ Alina Moldovan commented:

Verified as fixed using V9000 (35002) and iPhone 14+ (16.0.3).

Razvan Litianu Manage to set Fiddler to interrupt the network traffic on device and simulate a 500 status call response 

Here is what I got:

  1. Each request is sent twice (even the first response in 200) - since I’m not able to decode the request body I cannot be 100% sure that the two requests are identical 
  2. The request in sent only twice when the response code in 500. 

!A5526203-8913-4ADC-9B1B-718BD52FFD85.mov|width=1758,height=1080!
